### PR TITLE
Don't interpolate joboutputs, before job is done

### LINKED ISSUE #758: Issue: Only first step is able to set output variables
## System information

- Operating System: Windows 10
- Architecture: x64 (64-bit)
- Apple M1: no
- Docker version: 20.10.7
- `act` version: c865a56 (current commit on `master` at the time of submission)

## Expected behaviour

Output variables should be able to be defined in multiple steps under the same job

## Actual behaviour

Currently, output variables can only be successfully defined in the first step inside a job (see official [documentation](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs) for reference).

## Workflow and/or repository

<details>
  <summary>workflow</summary>

```yml
name: Test
on:
  workflow_dispatch:

jobs:
  set_vars:
    name: Set variables
    runs-on: ubuntu-latest
    steps:
    - id: set_1
      run: |
        echo "::set-output name=var_1::$(echo hello)"
        echo "::set-output name=var_2::$(echo world)"
    - id: set_2
      run: echo "::set-output name=var_3::$(echo 1234)"
    outputs:
      variable_1: ${{ steps.set_1.outputs.var_1 }}
      variable_2: ${{ steps.set_1.outputs.var_2 }}
      variable_3: ${{ steps.set_2.outputs.var_3 }}
  printing:
    name: Printing
    needs: set_vars
    runs-on: ubuntu-latest
    steps:
      - name: Print variables from another job
        run: |
          echo "${{ needs.set_vars.outputs.variable_1 }}" # Gets printed
          echo "${{ needs.set_vars.outputs.variable_2 }}" # As well
          echo "${{ needs.set_vars.outputs.variable_3 }}" # Empty space in logs, variable not resolved
```

</details>

## Steps to reproduce

Run: `act workflow_dispatch -W <path to the workflow file with content above>`

## `act` output

<details>
  <summary>Log</summary>

```none
[Test/Set variables] 🚀  Start image=catthehacker/ubuntu:act-latest
[Test/Set variables]   🐳  docker run image=catthehacker/ubuntu:act-latest platform= entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Test/Set variables]   🐳  docker exec cmd=[mkdir -m 0777 -p /var/run/act] user=root
[Test/Set variables] ⭐  Run echo "::set-output name=var_1::$(echo hello)"
echo "::set-output name=var_2::$(echo world)"
[Test/Set variables]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /mnt/c/testrepo/workflow/set_1] user=
[Test/Set variables]   ⚙  ::set-output:: var_1=hello
[Test/Set variables]   ⚙  ::set-output:: var_2=world
time="2021-07-20T21:21:08+02:00" level=error msg="Unable to interpolate string '${{ steps.set_2.outputs.var_3 }}' - [TypeError: Cannot access member 'outputs' of undefined]"
[Test/Set variables]   ✅  Success - echo "::set-output name=var_1::$(echo hello)"
echo "::set-output name=var_2::$(echo world)"
[Test/Set variables] ⭐  Run echo "::set-output name=var_3::$(echo 1234)"
[Test/Set variables]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /mnt/c/testrepo/workflow/set_2] user=
[Test/Set variables]   ⚙  ::set-output:: var_3=1234
[Test/Set variables]   ✅  Success - echo "::set-output name=var_3::$(echo 1234)"
[Test/Printing     ] 🚀  Start image=catthehacker/ubuntu:act-latest
[Test/Printing     ]   🐳  docker run image=catthehacker/ubuntu:act-latest platform= entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Test/Printing     ]   🐳  docker exec cmd=[mkdir -m 0777 -p /var/run/act] user=root
[Test/Printing     ] ⭐  Run Print variables from another job
[Test/Printing     ]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /mnt/c/testrepo/workflow/0] user=
| hello
| world
| 
[Test/Printing     ]   ✅  Success - Print variables from another job
```

</details>

## Comments

May relate to [issue 295](https://github.com/nektos/act/issues/295) based on the comments at the end, but this one is a different issue as `needs` now work on the master branch, although not released yet.

This can affect workflows where the first step pulls in an action, e.g. using the digitalocean/action-doctl@v2 as the first step in order to execute `doctl` commands that gets the IP address of a Droplet and sets it as the output in the second step will not be possible due to this issue.

### PR DESCRIPTION
**Pull request by BlackDex**
All credits go to @ChristopherHX which fixed this issue (https://github.com/ChristopherHX/act/commit/fd9f40cdc02dc72c40a9c33450d638d6e8e8ee57).
I only created a PR for this so it will be fixed in the upstream binary.

This fixes #758

### CODE REVIEW

**Comment by ChristopherHX** [comment]
Thank you for the backport. I didn't create a PR myself, due to merge conflicts with my own PR's at the same time.

**Comment by catthehacker** [comment]
Please add tests for it

**Comment by BlackDex** [comment]
@catthehacker I have added a test, hope that this is enough :)

**Comment by ChristopherHX** [comment]
@BlackDex You also need to add the test workflow to the list of this file https://github.com/nektos/act/blob/96cf907c5e1f4761ef6d91bb8e22487747a77f6b/pkg/runner/runner_test.go#L123

Otherwise it doesn't run during CI or go-test.

**Comment by BlackDex** [comment]
Ok, i have done that now lets wait and see.

File: pkg/runner/runner_test.go [hunk 1]
```diff
@@ -120,6 +120,7 @@ func TestRunEvent(t *testing.T) {
 		{"testdata", "issue-597", "push", "", platforms, ""},
 		{"testdata", "issue-598", "push", "", platforms, ""},
 		{"testdata", "env-and-path", "push", "", platforms, ""},
+		{"testdata", "output", "push", "", platforms, ""},
```
Comment (ChristopherHX) [inline pkg/runner/runner_test.go:123]: Typo?
Your workflow is in folder `outputs` not `output`.

Comment (BlackDex) [reply pkg/runner/runner_test.go]: Indeed

**Review by catthehacker — APPROVED** [review]

**Review by cplee — APPROVED** [review]

**Pull request merged**